pkg/scaffold/v2/prometheus/monitor.go

```diff
@@ -0,0 +1,37 @@
+package prometheus
+
+import (
+	"path/filepath"
+	"sigs.k8s.io/kubebuilder/pkg/scaffold/input"
+)
+
+// PrometheusMetricsService scaffolds an issuer CR and a certificate CR
+type PrometheusMonitorService struct {
+	input.Input
+}
+
+// GetInput implements input.File
+func (p *PrometheusMonitorService) GetInput() (input.Input, error) {
+	if p.Path == "" {
+		p.Path = filepath.Join("config", "prometheus", "monitor.yaml")
+	}
+	p.TemplateBody = monitorTemplate
+	return p.Input, nil
+}
+
+var monitorTemplate = `
+# Prometheus Monitor Service (Metrics)
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  labels:
+    control-plane: controller-manager
+  name: controller-manager-metrics-monitor
+  namespace: system
+spec:
+  endpoints:
+    - path: /metrics
+      port: https
```

Comment:
IIUC `ServiceMonitor` requires a service that defined a port named `https`.
Kubebuilder currently doesn't have a service for metrics.

Reply:
HI @mengqiy, 

Actually it has been creating in the project already by `pkg/scaffold/v2/authproxyservice.go`. So, in order to not duplicated it and not affect the auth feature, it is the service which will be used by the Monitor.   

I added in the description of the PR all steps performed to test it and there you can see that the service/monitor-controller-manager-metrics-service is created.

Please, let me know if I am missing something here.

Reply:
Surprised to discover the auth service is under rbac directory.
IMO it should not be in `rbac` directory, since that directory is for (Cluster)Role and (Cluster)RoleBinding.

Reply:
Wdyt about add it under the default one? I think it is the best place?

Reply:
Yeah, moving files around is refactoring. It is out-of-scope for this PR.
I was explaining why I miss that file earlier since the file is in an unexpected location. Sorry for the confusion.

Reply:
No worry and really tks for your time and review on it.